cshared_test.go

```diff
@@ -427,3 +431,195 @@ func TestInputCallbackInfiniteConcurrent(t *testing.T) {
 			concurrentCountFinish.Load())
 	}
 }
+
+type testOutputHandlerReflect struct {
+	param        string
+	flushCounter metric.Counter
+	log          Logger
+}
+
+func (plug *testOutputHandlerReflect) Init(ctx context.Context, fbit *Fluentbit) error {
+	plug.flushCounter = fbit.Metrics.NewCounter("flush_total", "Total number of flushes", "gstdout")
+	plug.param = fbit.Conf.String("param")
+	plug.log = fbit.Logger
+
+	return nil
+}
+
+func (plug *testOutputHandlerReflect) Flush(ctx context.Context, ch <-chan Message) error {
+	// Iterate Records
+	count := 0
+	printout := bytes.NewBuffer([]byte{})
+
+	for {
+		select {
+		case msg := <-ch:
+			rec := reflect.ValueOf(msg.Record)
+			printout.WriteString("[{")
+			if rec.Kind() == reflect.Map {
+				keyCount := 0
+				for _, key := range rec.MapKeys() {
+					if keyCount > 0 {
+						printout.WriteString(", ")
+					}
+					strct := rec.MapIndex(key)
+					printout.WriteString(
+						fmt.Sprintf("\"%s\":\"%v\"",
+							key.Interface(), strct.Interface()))
+					keyCount++
+				}
+			}
+			printout.WriteString("}]")
+			count++
+		case <-ctx.Done():
+			if count <= 0 {
+				return fmt.Errorf("no records flushed")
+			}
+			fmt.Print(printout.String())
```

Comment:
Shouldn't you be doing some assertions here instead of just printing the result?

Reply:
will add assertions.

Reply:
I've added assertions, one of them is pretty weird though:

```go
			foo, ok := record["foo"]
			if !ok {
				t.Errorf("unable to access value for foo")
				return nil
			}

			// this is a little bit disconcerting. does the msgpack
			// decoder default to using []uint8 for strings?
			if foostr, okfoo := foo.([]uint8); !okfoo {
				t.Errorf("unable to get value for foo")
				return nil
			} else if string(foostr) != "bar" {
				t.Errorf("invalid value for foo")
				return nil
			}

```

Apparently the decoder is favouring `[]uint8` over `string` (which is technically mostly accurate for C strings... but....).